horace_core/sqw/coord_transform/@spher_proj/private/transform_pix_to_spher_.m

```diff
@@ -0,0 +1,48 @@
+function pix_transf = transform_pix_to_spher_(obj,pix_input,varargin)
+% Transform pixels expressed in crystal Cartesian coordinate systems
+% into image coordinate system
+%
+% Input:
+% pix_data -- [3xNpix] or [4xNpix] array of pix coordinates
+%             expressed in crystal Cartesian coordinate system
+%             or pixelData object containing this information.
+% Returns:
+% pix_out -- [3xNpix or [4xNpix] Array the pixels coordinates transformed
+%            into spherical coordinate system defined by object properties
+%
+if isa(pix_input,'PixelDataBase')
+    pix_cc = pix_input.q_coordinates;
+    if obj.offset(4) ~=0
+        shift_ei = true;
+    else
+        shift_ei = false;
+    end
+    ndim = 3;
+    input_is_obj = true;
+else % if pix_input is 4-d, this will use 4-D matrix and shift
+    % if its 3-d -- matrix is 3-dimensional and energy is not shifted
+    % anyway
+    ndim         = size(pix_input,1);
+    pix_cc       = pix_input;    
+    input_is_obj = false;
+end
+
+[rot_mat,offset] = obj.get_pix_img_transformation(ndim);
+
+%
+pix_transf= ((bsxfun(@minus,pix_cc,offset'))'*rot_mat')';
```

Comment:
noting that matlab docs for bsxfun say you can do this elementwise from 2016b, so pix_cc.-offset. Is there still a good reason for using bsxfun?

Reply:
Edges ago we did testing and it was order of magnitude faster. I do not know if it still faster or we can replace it. Probably good technical dept question.  Unfortunately we do not have a unit test which is sufficiently good to replace it trivially.